cadquery/occ_impl/assembly.py

```diff
@@ -224,7 +224,10 @@ def _toCAF(el, ancestor, color) -> TDF_Label:
                 if coloredSTEP and current_color:
                     setColor(lab, current_color, ctool)
 
-            tool.AddComponent(subassy, lab, TopLoc_Location())
+            if ancestor:
```

Comment:
Same here, I'm not sure what this if/else actually does

Reply:
If there is an ancestor, no change - the location for this object is applied when adding to the higher level assy.

If no ancestor it means this object is a root object of the assembly.  Since this object is not added to any other higher level assembly, need to apply the top location.